src/components/Personalization/validateApplyPropositionsOptions.js

```diff
@@ -0,0 +1,40 @@
+import {
+  anything,
+  boolean,
+  string,
+  objectOf,
+  arrayOf
+} from "../../utils/validation";
+
+export default objectOf({
+  propositions: arrayOf(
+    objectOf({
+      renderAttempted: boolean(),
+      id: string(),
+      scope: string(),
+      items: arrayOf(
+        objectOf({
+          id: string(),
+          schema: string(),
+          meta: objectOf({
+            "experience.id": string(),
+            "activity.id": string(),
+            "offer.name": string(),
+            "activity.name": string(),
+            "offer.id": string()
+          }),
+          data: objectOf({
+            type: string(),
+            format: string(),
+            content: string(),
+            selector: string(),
+            prehidingSelector: string()
+          })
+        })
+      ),
+      scopeDetails: objectOf(anything())
+    })
+  ).nonEmpty(),
+  viewName: string(),
+  notificationsEnabled: boolean()
+}).required();
```

Comment:
I'm wondering if we need to provide such strict validation here when users will simply take the propositions from the sendEvent call and send them to applyPropositions. For example, when we support ODE offers, they may not be in the same format.